agent/engine/docker_container_engine.go

```diff
@@ -371,6 +336,50 @@ func (dg *dockerGoClient) pullImage(image string, authData *api.RegistryAuthenti
 	return nil
 }
 
+func (dg *dockerGoClient) filterDebugOutput(pullDebugOut *io.PipeReader, pullBegan chan bool, image string) {
```

Comment:
`filterPullDebugOutput` would be a more suitable name for this method. Also, since you're never reading from the  `pullBegan` channel and only sending/writing events to it, you can rewrite your method signature as `pullBegan<- chan bool` and make that more explicit. Example: https://gobyexample.com/channel-directions

Reply:
method renamed, and method signature updated (pullBegan chan<- bool).